puppet/chassis.rb

```diff
@@ -134,6 +134,20 @@ def self.normalize_config(config)
 					config["mapped_paths"][path] = config["mapped_paths"]["base"]
 				end
 			end
+
+			machine_name = '' unless config["machine_name"]
+
+			# If we don't have a machine name in the config generate one from the directory structure.
+			if machine_name == ''
+				pwd = Dir.pwd.downcase
+				name = pwd.gsub("/","_")
+				if name[0] == "/"
```

Comment:
Is this a weird side effect of how `gsub` works? Pretty annoying! Thanks for making a start on this btw

Reply:
Nah not at all. I was just following your comment:
>/Users/bob/chassis becomes users_bob_chassis
So I downcased it and needed to remove the first `/`